app/controllers/users/approvals_controller.rb

```diff
@@ -22,4 +26,15 @@ def reject
     render "users/approvals/approve"
   end
 
+  def check_user
+    unless current_user?(params[:user_id])
+      flash[:notice] = "You are signed in as #{current_user.username}"
+      if params[:origin]
+        developer = Developer.find_by(company_name: params[:company_name])
```

Comment:
@dansteele Couldn't find a way to add headers to the request issued by the redirect_to Coposition in the WFAT code. Couldn't find a way to replace the redirect_to with an HTTParty get request either so probably gunna stick with this.
@GarySiu merge this if it all looks ok.